yarn.lock

```diff
@@ -8760,7 +8690,7 @@ browserslist@4.10.0:
     node-releases "^1.1.52"
     pkg-up "^3.1.0"
 
-browserslist@^4.0.0, browserslist@^4.11.1, browserslist@^4.12.0, browserslist@^4.6.4, browserslist@^4.8.0, browserslist@^4.8.2, browserslist@^4.8.5, browserslist@^4.9.1:
```

Comment:
@scinos `browserslist` could use further deduplication here. Now it has two versions installed, 4.14.0 and 4.14.7. `caniuse-lite` and `autoprefixer` are in a similar situation.

Noticed this when resolving lockfile conflicts with #48729.

Reply:
I'm not sure it can be safely deduplicated, as our tree requires two different non-overlapping ranges of `browserlist`:

* `react-dev-utils@10.0.0` depends on `browserslist@4.10.0` (fixed version, not a range)
* Other packages depends on `browserslist@^4.11.1` or `browserslist@^4.12.0`.

Most likely `react-dev-utils` will work fine with `browserlist@4.14.0`, but I rather not override the version specified by a project without first understanding why `react-dev-utils` uses a fixed version and maybe asking them to switch to ranges.

Looks like this "explicit duplication" is very common in our repo: a basic grep tells me we have 548 packages with two or more versions, while `yarn-deduplicate --list` only detects 136 packages candidates to be deduplicated.

Reply:
I see, there are actually three records for `browserslist` in the lockfile. I haven't noticed the first, pinned one before:
1. `browserslist@4.10.0` -> `4.10.0`. That's a pinned version, nothing can be done about it.
2. `browserslist@^4.0.0 + 5 other specs` -> `4.14.0`. All of them use carets (`^`) and should be flexible.
3. `browserslist@^4.14.5` -> `4.14.7`. This one could be deduplicated and merged into the big list above. And the whole list should deduplicate into the 4.14.7 version.

I wonder why `yarn-deduplicate` didn't find that records 2 and 3 can be merged into one? Maybe it got confused by record 1 somehow?

Reply:
Very interesting! Maybe the extra record was added after I run the initial `yarn-deduplicate` for browserslist? 

It seems to detect it fine now in trunk:

```
Package "browserslist" wants ^4.0.0 and could get 4.14.7, but got 4.14.0
Package "browserslist" wants ^4.12.0 and could get 4.14.7, but got 4.14.0
Package "browserslist" wants ^4.6.4 and could get 4.14.7, but got 4.14.0
Package "browserslist" wants ^4.8.2 and could get 4.14.7, but got 4.14.0
Package "browserslist" wants ^4.8.5 and could get 4.14.7, but got 4.14.0
Package "browserslist" wants ^4.9.1 and could get 4.14.7, but got 4.14.0
```

And in fact, `npx yarn-deduplicate --packages browserslist` moves all specs using carets to one single version, `4.14.7`

Reply:
Actually it looks like a bad merge: the original PR correctly deduplicates all versions: https://github.com/Automattic/wp-calypso/pull/48594/files

Reply:
Interesting. In that `autoprefixer` deduplication, some `browserslists` semvers stopped being requested, so it makes sense that the `browserslists` record would be modified (removing an item from the list). But why it was split into two? Looking at the diff, that was not necessary at all.

Reply:
I did those in two different PRs (actually I have a script that does each dedupe in an individual PR), but I merge them into this branch/PR to avoid creating too much noise in the deployment queue.